backend/src/database/class_enrollments.go

```
@@ -190,28 +193,42 @@ func (db *DB) UpdateProgramClasses(ctx context.Context, classIDs []int, classMap
 		return newUpdateDBError(tx.Error, "begin transaction")
 	}
 
-	if status, ok := classMap["status"]; ok &&
-		status == string(models.Cancelled) || status == string(models.Completed) {
+	if rawStatus, ok := classMap["status"]; ok {
+		var statusStr string
+		switch v := rawStatus.(type) {
+		case string:
+			statusStr = v
+		case models.ClassStatus:
+			statusStr = string(v)
+		default:
+			statusStr = fmt.Sprint(v)
+		}
 
-		for _, classID := range classIDs {
-			var enrollmentStatus models.ProgramEnrollmentStatus
-			switch status {
-			case string(models.Cancelled):
-				enrollmentStatus = models.EnrollmentCancelled
-			case string(models.Completed):
-				enrollmentStatus = models.EnrollmentCompleted
+		if statusStr == string(models.Cancelled) || statusStr == string(models.Completed) {
+			for _, classID := range classIDs {
+				var enrollmentStatus models.ProgramEnrollmentStatus
+				switch statusStr {
+				case string(models.Cancelled):
+					enrollmentStatus = models.EnrollmentCancelled
+				case string(models.Completed):
+					enrollmentStatus = models.EnrollmentCompleted
+				}
+				if err := tx.
+					Model(&models.ProgramClassEnrollment{}).
+					Where("class_id = ? AND enrollment_status = ?", classID, models.Enrolled).
+					Set("class_id", classID).
+					Update("enrollment_status", enrollmentStatus).
+					Error; err != nil {
+					tx.Rollback()
+					return newUpdateDBError(err, "class enrollment statuses")
+				}
 			}
-			if err := tx.
-				Model(&models.ProgramClassEnrollment{}).
-				Where("class_id = ? AND enrollment_status = ?", classID, models.Enrolled).
-				Set("class_id", classID).
-				Update("enrollment_status", enrollmentStatus).
-				Error; err != nil {
+
+			if err := db.updateRecurrenceRulesForClassCompletion(tx, classIDs); err != nil {
 				tx.Rollback()
-				return newUpdateDBError(err, "class enrollment statuses")
+				return err
```

Need to return the appropriate error here.